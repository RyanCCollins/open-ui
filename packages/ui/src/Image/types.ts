import * as React from 'react';
import Image from './';

export type ImageSize = 'thumb' | 'xxsmall' | 'xsmall' | 'small' | 'medium' | 'large' | 'xlarge' | 'full';

export interface Props extends React.Props<typeof Image> {
  size?: ImageSize;
  src: string;
  alt: string;
}
