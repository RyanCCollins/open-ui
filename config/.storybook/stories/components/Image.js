import React from 'react';
import { storiesOf, action } from '@kadira/storybook';
import { Image } from 'ui';
import {
  withKnobs,
  select
} from '@kadira/storybook-addon-knobs';

const sizeOptions = {
  thumb: 'thumb',
  xxsmall: 'xxsmall',
  xsmall: 'xsmall',
  small: 'small',
  medium: 'medium',
  large: 'large',
  full: 'full'
};

const srcUrl='https://goo.gl/zm8EZD';

const stories = storiesOf('Image', module);
stories.addDecorator(withKnobs);

stories
  .addWithInfo(
    'Image',
    `
      Properties: 

      \`size: thumb|xxsmall|xsmall|small|medium|large|full\`
      
      Usage:

      ~~~js
        <Image
          size='small'
          alt='alt text'
          src='/src/to/image'
        />
      ~~~
    `,
     () => (
        <Image
          size={select('size', sizeOptions, 'thumb')}
          alt='tree shaking'
          src={srcUrl}
        />
     ),
     {
       inline: true,
       propTables: false,
     }
  );
