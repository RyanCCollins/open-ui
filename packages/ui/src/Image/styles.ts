import remStringFromPx from '../utils';
import { Props } from './types';
const styled = require('styled-components').default;
const { css } = require('styled-components');

const imageSizeMap = {
  thumb: 50,
  xxsmall: 100,
  xsmall: 200,
  small: 300,
  medium: 400,
  large: 500,
  xlarge: 600,
  full: 1200,
};

export const imageCss = ({ size }: Props) => {
  const sizeWithDefault = size || 'small';
  const px = imageSizeMap[sizeWithDefault];
  const rem = remStringFromPx(px);
  if (sizeWithDefault === 'full') {
    return css`
      width: 100vw;
      height: auto;
    `;
  }
  if (sizeWithDefault === 'thumb') {
    return css`
      width: ${rem};
      height: ${rem};
      flex: 0 0 auto;
      object-fit: cover;
    `;
  }
  return css`
    max-width: 100%;
    height: auto;
    min-height: ${rem};
    max-height: ${rem};
    display: block;
  `;
};

export const Img = styled.img`
  ${(props) => imageCss(props)}
`;
