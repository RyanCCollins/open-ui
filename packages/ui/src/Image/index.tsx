import * as React from 'react';
import { Img } from './styles';
import { Props } from './types';

export default function Image({
  ...props,
}: Props): JSX.Element {
  return(
    <Img {...props} />
  );
};
