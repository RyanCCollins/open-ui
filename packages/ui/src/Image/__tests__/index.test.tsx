import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import * as React from 'react';
import Image from '../';
import * as Props from './__mocks__/imageMocks.mock';

describe('<Image />', () => {
  it('should render a default image', () => {
    const wrapper = shallow(
      <Image {...Props.MockImageProps} />,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
  it('should render a thumbnail image', () => {
    const wrapper = shallow(
      <Image {...Props.MockImagePropsThumb} />,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
  it('should render a xxsmall image', () => {
    const wrapper = shallow(
      <Image {...Props.MockImagePropsXXSmall} />,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
  it('should render a xsmall image', () => {
    const wrapper = shallow(
      <Image {...Props.MockImagePropsXSmall} />,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
  it('should render a small image', () => {
    const wrapper = shallow(
      <Image {...Props.MockImagePropsSmall} />,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
  it('should render a medium image', () => {
    const wrapper = shallow(
      <Image {...Props.MockImagePropsMedium} />,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
  it('should render a large image', () => {
    const wrapper = shallow(
      <Image {...Props.MockImagePropsLarge} />,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
  it('should render a xlarge image', () => {
    const wrapper = shallow(
      <Image {...Props.MockImagePropsXLarge} />,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
  it('should render a full image', () => {
    const wrapper = shallow(
      <Image {...Props.MockImagePropsFull} />,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
