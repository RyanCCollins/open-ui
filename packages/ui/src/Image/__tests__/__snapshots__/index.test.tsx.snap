// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Image /> should render a default image 1`] = `
<styled.img
  alt="Some alt text"
  src="A source string"
/>
`;

exports[`<Image /> should render a full image 1`] = `
<styled.img
  alt="Some alt text"
  size="full"
  src="A source string"
/>
`;

exports[`<Image /> should render a large image 1`] = `
<styled.img
  alt="Some alt text"
  size="large"
  src="A source string"
/>
`;

exports[`<Image /> should render a medium image 1`] = `
<styled.img
  alt="Some alt text"
  size="medium"
  src="A source string"
/>
`;

exports[`<Image /> should render a small image 1`] = `
<styled.img
  alt="Some alt text"
  size="small"
  src="A source string"
/>
`;

exports[`<Image /> should render a thumbnail image 1`] = `
<styled.img
  alt="Some alt text"
  size="thumb"
  src="A source string"
/>
`;

exports[`<Image /> should render a xlarge image 1`] = `
<styled.img
  alt="Some alt text"
  size="xlarge"
  src="A source string"
/>
`;

exports[`<Image /> should render a xsmall image 1`] = `
<styled.img
  alt="Some alt text"
  size="xsmall"
  src="A source string"
/>
`;

exports[`<Image /> should render a xxsmall image 1`] = `
<styled.img
  alt="Some alt text"
  size="xxsmall"
  src="A source string"
/>
`;
